src/core/thread/discover_scanner.cpp

```diff
@@ -144,6 +166,23 @@ otError DiscoverScanner::Discover(const Mac::ChannelMask &aScanChannels,
     return error;
 }
 
+otError DiscoverScanner::SetJoinerAdvertisement(uint32_t aOui, const uint8_t *aAdvData, uint8_t aAdvDataLength)
+{
+    otError error = OT_ERROR_NONE;
+
+    VerifyOrExit((aAdvData != nullptr) && (aAdvDataLength != 0) &&
```

Comment:
Should we allow `aAdvLength == 0` to allow the user to clear the advertisement data?

Reply:
I'm a bit hesitated to allow the user to clear the advertisement data as this may bring in some additional implications to border agent and commissioners. Maybe we can introduce an enhancement later when we have better understanding on the use cases?